src/mlpack/core/data/string_cleaning.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file string_cleaning.hpp
+ * @author jeffin sam
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRING_CLEANING_HPP
+#define MLPACK_CORE_DATA_STRING_CLEANING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <boost/range/algorithm/remove_if.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <unordered_set>
+
+namespace mlpack {
+namespace data {
+
+class string_cleaning
+{
+ public:
+  /**
+  * Function to remove punctuation from a given vector of strings.
+  *
+  * @param input Vector of strings
+  */
+  void RemovePunctuation(std::vector<std::string>& input)
```

Comment:
It may be worth considering adding `std::string punctuation` as a parameter with default value `"!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}-"`, so that the user can specify their own set of punctuation characters, if perhaps there are some they would like to keep.

Maybe it is better instead to pass a function here like `std::isalnum()` or something like that, where if the function returns true then the character is kept?  (Or something kind of like that.)

Reply:
I think `std::ispunct()` suits better than `std::isalnum()`. I agree we could pass either a string or a function.